### PR TITLE
docs: Daemonset sharding example

### PR DESCRIPTION
**Pull request by CatherineF-dev**
**What this PR does / why we need it**: provide scaling mode example for https://github.com/kubernetes/kube-state-metrics/pull/1864 

**How does this change affect the cardinality of KSM**: *(increases, decreases or does not change cardinality)*

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:
Fixes https://github.com/kubernetes/kube-state-metrics/issues/1863

### CODE REVIEW

**Comment by CatherineF-dev** [comment]
Also need to add a deployment which watches other resources.

File: examples/standard/deployment.yaml [hunk 1]
```diff
@@ -1,5 +1,5 @@
 apiVersion: apps/v1
-kind: Deployment
+kind: Daemonset
```
Comment (mrueg) [inline examples/standard/deployment.yaml:2]: I believe this should probably stay a deployment?

**Comment by fpetkovski** [comment]
Do we generate examples from jsonnet, or are they managed separately? I noticed that we use an older image version and keeping the examples up to date will need some work if they're not auto-generated.

File: examples/standard/deployment.yaml (hunk 1 above)
Comment (CatherineF-dev) [reply examples/standard/deployment.yaml:2]: Yes, I was testing that. I should add a TODO(revert to Deployment after testing) there.

**Comment by fpetkovski** [comment]
Thanks, looks great so far. I just wonder if it's possible to name the folder `daemonset-sharding` instead of `daemonsetSharding` to consistency with file names. Not a blocker though, I understand this can be a limitation of how we generate examples from jsonnet.

**Comment by mrueg** [comment]
If we don't generate it with jsonnet, will container image versions need to be updated manually?

**Comment by CatherineF-dev** [comment]
Yes, it has a TODO (use jsonnet to generate yaml) in [previous comment](https://github.com/kubernetes/kube-state-metrics/pull/1873#issuecomment-1418350442). Will add jsonnet in this PR.

The generated Yamls are similar with/without jsonnet.

**Comment by CatherineF-dev** [comment]
Have fixed presubmits, ready for review~

**Comment by mrueg** [comment]
/lgtm

Thanks @CatherineF-dev

**Pull request merged**